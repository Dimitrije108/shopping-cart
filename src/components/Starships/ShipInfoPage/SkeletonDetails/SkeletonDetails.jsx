import styles from "./SkeletonDetails.module.css"
// Create a ship card component
export default function SkeletonDetails() {
	return (
		<div className={styles.skelPage}  data-testid="skelDetails">
			<div className={styles.skelImg}></div>
			<div  className={styles.descCont}>
				<div className={styles.name}></div>
				<div className={styles.description}></div>
			</div>
			<div className={styles.price}></div>
			<div  className={styles.tableCont}>
				<div className={styles.spec}></div>
				<div className={styles.skelTable}>
					<div className={styles.table}></div>
					<div className={styles.table}></div>
				</div>
			</div>
		</div>
	)
};