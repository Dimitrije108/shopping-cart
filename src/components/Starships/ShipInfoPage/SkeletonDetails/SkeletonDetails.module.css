.skelPage {
	width: 1100px;
	padding: 0.5rem;

	display: grid;
	grid-template-columns: 3fr 1fr;
	grid-template-rows: auto;
	grid-template-areas: 
		"img img"
		"description price"
		"table price";
	gap: 2rem;
}

.skelImg {
	height: 500px;
  margin-bottom: 1rem;
	outline: 1px solid rgb(104, 104, 104);
	
	grid-area: img;
}

.descCont {
	grid-area: description;
}

.name {
	width: 40%;
	height: 35px;
	margin-bottom: 1rem;
}

.description {
	width: 100%;
	height: 200px;
	margin: 0.5rem 0;
}

.price {
	height: 350px;
	outline: 1px solid rgb(104, 104, 104);

	grid-area: price;
}

.tableCont {
	grid-area: table;
}

.spec {
	width: 40%;
	height: 30px;
	margin-bottom: 1rem;
}

.skelTable {
	width: 100%;
	height: 250px;
	display: flex;
	gap: 1rem;
}

.table {
	width: 50%;
	outline: 1px solid rgb(104, 104, 104);
}

.skelImg,
.name,
.description,
.price,
.spec,
.table {
  background: linear-gradient(90deg, #141414 25%, #1d1d1d 50%, #131313 75%);
  background-size: 200% 100%;
  animation: shimmer 5s infinite;
}

@keyframes shimmer {
  from {
    background-position: 200% 0;
  }

  to {
    background-position: -200% 0;
  }
}