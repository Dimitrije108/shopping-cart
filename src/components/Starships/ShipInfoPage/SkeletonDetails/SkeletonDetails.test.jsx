import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import SkeletonDetails from "./SkeletonDetails"

describe("Skeleton Details component", () => {
	it("skeleton details component snapshot", () => {
		const skeletonDetails = render(
			<MemoryRouter>
				<SkeletonDetails />
			</MemoryRouter>
		)

		expect(skeletonDetails).toMatchSnapshot();
	});
});