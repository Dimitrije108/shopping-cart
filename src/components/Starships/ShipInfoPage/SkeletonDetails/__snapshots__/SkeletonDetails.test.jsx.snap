// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skeleton Details component > skeleton details component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_skelPage_56ff8f"
        data-testid="skelDetails"
      >
        <div
          class="_skelImg_56ff8f"
        />
        <div
          class="_descCont_56ff8f"
        >
          <div
            class="_name_56ff8f"
          />
          <div
            class="_description_56ff8f"
          />
        </div>
        <div
          class="_price_56ff8f"
        />
        <div
          class="_tableCont_56ff8f"
        >
          <div
            class="_spec_56ff8f"
          />
          <div
            class="_skelTable_56ff8f"
          >
            <div
              class="_table_56ff8f"
            />
            <div
              class="_table_56ff8f"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_skelPage_56ff8f"
      data-testid="skelDetails"
    >
      <div
        class="_skelImg_56ff8f"
      />
      <div
        class="_descCont_56ff8f"
      >
        <div
          class="_name_56ff8f"
        />
        <div
          class="_description_56ff8f"
        />
      </div>
      <div
        class="_price_56ff8f"
      />
      <div
        class="_tableCont_56ff8f"
      >
        <div
          class="_spec_56ff8f"
        />
        <div
          class="_skelTable_56ff8f"
        >
          <div
            class="_table_56ff8f"
          />
          <div
            class="_table_56ff8f"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
